src/pseudopeople/noise_functions.py

```diff
@@ -148,21 +148,51 @@ def miswrite_ages(
 
 
 def miswrite_numerics(
-    form_data: pd.DataFrame,
-    configuration: float,
+    column: pd.Series,
+    configuration: ConfigTree,
     randomness_stream: RandomnessStream,
     additional_key: Any,
-) -> pd.DataFrame:
+) -> pd.Series:
     """
+    Function that noises numeric characters in a series.
 
-    :param form_data:
-    :param configuration:
-    :param randomness_stream:
+    :param column: A pd.Series
+    :param configuration: ConfigTree object containing noise level
+    :param randomness_stream: RandomnessStream for CRN framework.
     :param additional_key: Key for RandomnessStream
-    :return:
+
+    returns: pd.Series with some numeric values experiencing noise.
     """
-    # todo actually duplicate rows
-    return form_data
+
+    token_noise_level = configuration.token_noise_level
```

Comment:
There is a note in the docs that a nice-to-have would be to replace tokens with other digits (i.e. not replace a digit with itself). We can actually manage this quite simply by just multiplying the `token_level_noise` by 1/0.9 . I think this is worth doing,

Reply:
Correct, but consider this if `n` is the desired token level noise. Before we were noising `n * len(column)` rows, but only 90% of these actually changed (since we could replace with the same digit, so we were actually changing the value of `0.9 * n * len(column)` rows. If we change our implementation so we "noise" `n * len(column) / 0.9` rows, we will actually change the value of `0.9 * n * len(column) / 0.9 == n * len(column)` since the 0.9s cancel out.